test/publish.js

```diff
@@ -240,3 +257,32 @@ test('Publish dashboard - success w/ custom timeout', function t(assert) {
         timeout: 2000
     });  // 200
 });
+
+test('Publish dashboard - passes headers', function t(assert) {
+    config.configure({
+        cookie: cookie,
+        headers: headers,
+        url: url
+    });
+    // hijack the calls to elastic search, need to test both response codes
+    // since the initial request will return a 201 and the subsequent will
+    // return a 200.
+    nock(baseUrl)
+        .post('/dashboard')
+        .reply(201, function createdHandler() {
+            var reqHeaders = this.req.headers;
+            Object.keys(headers).forEach(function(key) {
+                assert.equal(reqHeaders[key], headers[key]);
+            });
+        })
+        .post('/dashboard')
+        .reply(200, function okHandler() {
+            var reqHeaders = this.req.headers;
+            Object.keys(headers).forEach(function(key) {
+                assert.equal(reqHeaders[key], headers[key]);
+            });
+            assert.end();
+        });
+    publish(dashboard); // 201
```

Comment:
Do we need timeout like line254?

Reply:
I'm just following line 225.  Timeout should be optional.